tests/test_integration.py

```diff
@@ -32,262 +27,107 @@
-    project_network = Query({'hostname': VM_NET}, ['intern_ip'] ).get()
-
-    try:
-        subnets = project_network['intern_ip'].subnets(subnet_levels)
-    except ValueError:
-        raise Exception(
-            ('Can\'t split {} into enough subnets '
-            'for {} parallel tests').format(vm_net, PYTEST_XDIST_WORKER_COUNT)
-        )
-
-    return [s for s in subnets][PYTEST_XDIST_WORKER][index]
-
-
 class IGVMTest(TestCase):
     def setUp(self):
         """Initialize VM object before every test
 
         Get object from Serveradmin and initialize it to safe defaults.
         Don't assign VM to any of HVs yet!
         """
-        # igvm operates always on hostname of VM and queries it from
-        # Serveradmin whenever it needs. Because of that we must never store
-        # any igvm objects and query things anew each time.
-        obj = Query({'hostname': VM_HOSTNAME}, [
-            'hostname',
-            'state',
-            'backup_disabled',
-            'disk_size_gib',
-            'memory',
-            'num_cpu',
-            'os',
-            'environment',
-            'no_monitoring',
-            'hypervisor',
-            'repositories',
-            'puppet_environment',
-            'puppet_ca',
-        ]).get()
-
-        # Fill in defaults in Serveradmin
-        obj['state'] = 'online'
-        obj['disk_size_gib'] = 3
-        obj['memory'] = 2048
-        obj['num_cpu'] = 2
-        obj['os'] = 'stretch'
-        obj['environment'] = 'testing'
-        obj['no_monitoring'] = True
-        obj['hypervisor'] = None
-        obj['repositories'] = [
+        super().setUp()
+
+        # Check that enough HVs are available.
+        self.route_network = Query(
+            {'hostname': VM_NET},
+            ['route_network'],
+        ).get()['route_network']
+
+        self.hvs = [Hypervisor(o) for o in Query({
+            'environment': 'testing',
+            'servertype': 'hypervisor',
+            'state': 'online',
+            'vlan_networks': self.route_network,
+        }, HYPERVISOR_ATTRIBUTES)]
+
+        assert len(self.hvs) >= 2, 'Not enough testing hypervisors found'
+
+        # Cleanup all leftovers from previous tests or failures.
+        clean_all(self.route_network, VM_HOSTNAME)
+
+        # Create subject VM object
+        self.vm_obj = Query().new_object('vm')
+        self.vm_obj['backup_disabled'] = True
+        self.vm_obj['disk_size_gib'] = 3
+        self.vm_obj['environment'] = 'testing'
+        self.vm_obj['hostname'] = VM_HOSTNAME
+        self.vm_obj['hypervisor'] = None
+        self.vm_obj['intern_ip'] = get_next_address(VM_NET, 1)
+        self.vm_obj['memory'] = 2048
+        self.vm_obj['no_monitoring'] = True
+        self.vm_obj['num_cpu'] = 2
+        self.vm_obj['os'] = 'stretch'
+        self.vm_obj['project'] = 'test'
+        self.vm_obj['puppet_environment'] = None
+        self.vm_obj['repositories'] = [
             'int:basestretch:stable',
```

Comment:
Isn't it better to start testing with basebuster?

Reply:
i was also surprised. but lets change this in another pr